client/matching/loadbalancer_test.go

```diff
@@ -0,0 +1,170 @@
+package matching
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/uber/cadence/common/dynamicconfig"
+	"github.com/uber/cadence/common/types"
+	"testing"
+)
+
+func TestNewLoadBalancer(t *testing.T) {
+	type args struct {
+		domainIDToName func(string) (string, error)
+		dc             *dynamicconfig.Collection
+	}
+	tests := []struct {
+		name string
+		args args
+		want LoadBalancer
+	}{
+		// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, NewLoadBalancer(tt.args.domainIDToName, tt.args.dc), "NewLoadBalancer(%v, %v)", tt.args.domainIDToName, tt.args.dc)
+		})
+	}
+}
+
+func Test_defaultLoadBalancer_PickReadPartition(t *testing.T) {
+	type fields struct {
+		nReadPartitions  dynamicconfig.IntPropertyFnWithTaskListInfoFilters
+		nWritePartitions dynamicconfig.IntPropertyFnWithTaskListInfoFilters
+		domainIDToName   func(string) (string, error)
+	}
+	type args struct {
+		domainID      string
+		taskList      types.TaskList
+		taskListType  int
+		forwardedFrom string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   string
+	}{
+		// TODO: Add test cases.
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			lb := &defaultLoadBalancer{
+				nReadPartitions:  tt.fields.nReadPartitions,
+				nWritePartitions: tt.fields.nWritePartitions,
+				domainIDToName:   tt.fields.domainIDToName,
+			}
+			assert.Equalf(t, tt.want, lb.PickReadPartition(tt.args.domainID, tt.args.taskList, tt.args.taskListType, tt.args.forwardedFrom), "PickReadPartition(%v, %v, %v, %v)", tt.args.domainID, tt.args.taskList, tt.args.taskListType, tt.args.forwardedFrom)
+		})
+	}
+}
+
+func Test_defaultLoadBalancer_PickWritePartition(t *testing.T) {
+	type fields struct {
+		nReadPartitions  dynamicconfig.IntPropertyFnWithTaskListInfoFilters
+		nWritePartitions dynamicconfig.IntPropertyFnWithTaskListInfoFilters
+		domainIDToName   func(string) (string, error)
+	}
+	type args struct {
+		domainID      string
+		taskList      types.TaskList
+		taskListType  int
+		forwardedFrom string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
```

Comment:
it's better to have expectedMax or expectedLists in the test struct. I know you can infer from nReadPartitions but this is more straightforward to have it explicitly mentioned here.